litellm/proxy/litellm_pre_call_utils.py

```diff
@@ -464,7 +464,22 @@ async def add_litellm_data_to_request(  # noqa: PLR0915
 
     from litellm.proxy.proxy_server import llm_router, premium_user
 
-    safe_add_api_version_from_query_params(data, request)
```

Comment:
add this logic within `safe_add_api_version_from_query_params` 

- first check if the user dynamically passed an api version 
- then read from llm_router (not from proxy_config)